components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/internal/handler/UserOperationEventHookHandler.java

```diff
@@ -99,7 +106,30 @@ public void handleEvent(Event event) throws IdentityEventException {
                     userOperationEventPublisherConfig.isPublishEnabled()) {
                 eventPayload = payloadBuilder.buildUserGroupUpdateEvent(eventData);
                 eventUri =
-                        eventConfigManager.getEventUri(Constants.EventHandlerKey.WSO2.POST_UPDATE_USER_LIST_OF_ROLE_EVENT);
+                        eventConfigManager.getEventUri(
+                                Constants.EventHandlerKey.WSO2.POST_UPDATE_USER_LIST_OF_ROLE_EVENT);
+                SecurityEventTokenPayload securityEventTokenPayload = EventHookHandlerUtils
+                        .buildSecurityEventToken(eventPayload, eventUri);
+                EventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+            } else if (IdentityEventConstants.Event.PRE_DELETE_USER_WITH_ID.equals(event.getEventName()) &&
+                    userOperationEventPublisherConfig.isPublishEnabled()) {
+
+                String userId =
+                        (String) event.getEventProperties().get(IdentityEventConstants.EventProperty.USER_ID);
+                // Setting the thread-local to keep user-ID for use when publishing post delete user event.
+                IdentityUtil.threadLocalProperties.get().put(PRE_DELETE_USER_USER_ID_FOR_WEB_SUB_HUB, userId);
```

Comment:
from where does this gets cleared ? however looks like it doesn't overall impacts as it gets set every time

Reply:
Fixed 8d8f0ebd9e807c0ac81c21e5a19eec711c59195f